client/valor/schemas/symbolic/types.py

```diff
@@ -2116,11 +2135,71 @@ def _get_type_by_value(other: typing.Any):
         raise NotImplementedError(str(type(other).__name__))
 
 
+def _parse_type_name(type_name: str):
+    """
+    Parses the type string of a variable into component types.
+
+    Parameters
+    ----------
+    type_name : str
+        The type name string to parse, which may contain nested types denoted by brackets.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the base type and its nested types as separate elements. The nested types are returned
+        as a single string if further nesting exists.
+
+    Notes
+    -----
+    The function only decomposes the string to the first level of nested types explicitly. For types with more
+    complex nested structures, the innermost structures are returned as part of the last component in the tuple,
+    without separating them further.
+
+    Examples
+    --------
+    >>> parse_type_name("a")
+    ('a',)
+
+    >>> parse_type_name("a[b]")
+    ('a', 'b',)
+
+    >>> parse_type_name("a[b[c]]")
+    ('a', 'b[c]',)
+
+    >>> parse_type_name("a[b[c[d]]]")
+    ('a', 'b[c[d]]',)
+    """
+    pattern = r"(\w+)(?:\[(\w+\[?\w*\]?)\])?"
+    matches = re.findall(pattern, type_name)
+    return tuple([str(item) for tup in matches for item in tup if item])
+
+
 def get_type_by_name(
-    name: str, additional_types: typing.Optional[typing.Dict[str, type]] = None
+    name: str, additional_types: typing.Optional[typing.List[type]] = None
 ):
-    """Retrieves variable type by name."""
-    types_ = {
+    """
+    Retrieves a class by its name.
```

Comment:
can you add more details here about how this is used?  the use of `additional_types` is hard to follow